internal/bminventory/inventory.go

```diff
@@ -3182,16 +3182,15 @@ func (b *bareMetalInventory) updateHostNames(ctx context.Context, params install
 			log.WithError(err).Errorf("invalid hostname format: %s", err)
 			return err
 		}
+		localhost = localhost || host.RequestedHostname == "localhost"
```

Comment:
What is the logic supposed to be here?

It looks like we're going to get `localhost == true` here for every host in the loop after the first one that is set to `localhost`.
That doesn't seem right.

Reply:
This is exactly what I aimed for: to have at least one host that updated its name from localhost to something else

Reply:
> to have at least one host that updated its name from localhost to something else

That isn't what this code is doing. I think you need to change the condition to `host.RequestedHostname != "localhost"`. And also ideally change the variable name to something more descriptive like `haveCustomHostname`.

Right now this variable is effectively `haveHostWithLocalhost` which I don't think is what you want.